src/components/Photo/DocumentLiveCapture.js

```diff
@@ -0,0 +1,160 @@
+  documentType: string,
+  className: string,
+  containerClassName: string,
+  renderTitle: Function,
+  renderError: Function
+}
+
+export default class DocumentLiveCapture extends Component<Props, State> {
+  webcam = null
+  snapshotIntervalRef: ?IntervalID = null
+
+  state: State = {
+    hasBecomeInactive: false,
+    hasCameraError: false,
+    snapshotBuffer: [],
+  }
+
+  handleTimeout = () => this.setState({ hasBecomeInactive: true })
+
+  handleCameraError = () => this.setState({ hasCameraError: true })
+
+  inactiveError = () => ({
+    name: this.props.isUploadFallbackDisabled ? 'CAMERA_INACTIVE_NO_FALLBACK' : 'CAMERA_INACTIVE',
+    type: 'warning'
+  })
+
+  captureDocument = (blob: Blob, sdkMetadata: Object) => {
+    const document = {
+      blob,
+      sdkMetadata,
+      filename: `document_capture.${mimeType(blob)}`
+    }
+    /* Attempt to get the 'ready' snapshot. But, if that fails, try to get the fresh snapshot - it's better
+       to have a snapshot, even if it's not an ideal one */
+    const snapshot = this.state.snapshotBuffer[0] || this.state.snapshotBuffer[1]
+    const captureData = { snapshot, ...document }
+    this.props.onCapture(captureData)
+  }
+
+  addSnapshotToBuffer = (blob: Blob, sdkMetadata: Object) => {
+    // Always try to get the older snapshot to ensure
+    // it's different enough from the user initiated capture
+    this.setState(({ snapshotBuffer: [, newestSnapshot] }) => ({
+      snapshotBuffer: [newestSnapshot, { blob, sdkMetadata, filename: `document_snapshot.${mimeType(blob)}` }]
+    }))
+  }
+
+  takeSnapshot = () => {
+    this.webcam && screenshot(this.webcam, this.addSnapshotToBuffer)
+  }
+
+  captureDocumentPhoto = () => screenshot(this.webcam, this.captureDocument)
+
+  startTakingSnapshots = () => {
+    sendEvent('Starting Live Document Capture')
+    const snapshotInterval = 1000
+    setTimeout(this.takeSnapshot, snapshotInterval / 4)
+    this.snapshotIntervalRef = setInterval(
+      this.takeSnapshot,
+      snapshotInterval
+    )
+  }
+
+  stopTakingSnapshots = () => {
+    if (this.snapshotIntervalRef) {
+      clearInterval(this.snapshotIntervalRef)
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.hasBecomeInactive) {
+      this.stopTakingSnapshots()
+    }
+  }
+
+  componentWillUnmount() {
+    this.stopTakingSnapshots()
+  }
+
+  render() {
+    const {
+      translate,
+      trackScreen,
+      renderFallback,
+      isUploadFallbackDisabled,
+      className,
+      containerClassName,
+      renderTitle,
+      renderError,
+      documentType
+    } = this.props
+    const { hasBecomeInactive, hasCameraError } = this.state
+    const id1SizeDocuments = new Set([ 'driving_licence', 'national_identity_card' ])
+    const documentSize = id1SizeDocuments.has(documentType) ? 'id1Card' : 'id3Card'
+    const heightAt1080p = 1080
+    return (
+      <div>
+        <Camera
+          facing='environment'
+          idealCameraHeight={ heightAt1080p }
```

Comment:
I'm not sure how this value would behave across multiple devices. We had a lot of issues when setting `height` and `width` constraints for `react-webcam`. Different browsers/devices behave in different ways. After several tests we found that certain height width combination break more often than other combination. Initially, we used higher resolution but for certain browsers the combination was failing when `getUserMedia` was called. We added props called `fallbackHeight` and `fallbackWidth` but that would trigger multiple user permissions pop-ups, which is not a nice experience. Therefore, we have been setting height to 720 and width is set automatically. In case this fails, we remove the height and width constraints because not setting any parameter works better than using `null` or a potentially failing value. More details [here](https://github.com/onfido/react-webcam/pull/22) and [here](https://github.com/onfido/react-webcam/pull/23)

Reply:
The image captured by the browser should be the closest value match to the required value capable. But it sounds like trying to increase resolution size by adjusting the height isn't a good idea reading your comment and the linked conversations. 

I did also notice that the version on the Onfido fork is a much older version than what's on the original repo (0.1.12 vs 2.0.0) and wondered if it was worth updating. But given the changes on our fork I have a feeling it might take a bit of comparing to see what's changed and if our workarounds are still needed.

Reply:
After looking through the code latest version of react-webcam, have added handling of screenshot quality as that was easy to add on to our existing code (it is actually an optional parameter in canvas.toBlob()).